### PR TITLE
[Modernize Code] fixed several issues reported by sonar

### PR DESCRIPTION
**Pull request by azoitl**
I played a bit with the output of the new sonar scanner and fixed reported issues in several places. 

@ptziegler I hope I didn't overdo it.

### CODE REVIEW

**Review by ptziegler — COMMENTED** [review]
Looking good so far. Nothing that really sticks out. I have some minor comments but that's about it.

File: org.eclipse.zest.core/.refactorings/2023/12/50/refactorings.history
[the remark this answers is not in the document]
Comment (azoitl) [reply org.eclipse.zest.core/.refactorings/2023/12/50/refactorings.history:1]: Sorry I think I shouldn't have committed this. Will remove it.

File: org.eclipse.draw2d/src/org/eclipse/draw2d/Label.java [hunk 1]
```diff
@@ -83,44 +83,34 @@ public Label(String s, Image i) {
...
 			alignOnHeight(textLocation, getTextSize(), textAlignment);
 			alignOnHeight(iconLocation, getIconSize(), iconAlignment);
 			break;
-		case NORTH:
-		case SOUTH:
+		case NORTH, SOUTH:
 			alignOnWidth(textLocation, getSubStringTextSize(), textAlignment);
 			alignOnWidth(iconLocation, getIconSize(), iconAlignment);
 			break;
+		default:
```
[the remark this answers is not in the document]
Comment (azoitl) [reply org.eclipse.draw2d/src/org/eclipse/draw2d/Label.java:112]: Draw2d is currently not using an logging framework. Not sure if we should add for that now one.

File: org.eclipse.zest.core/.refactorings/2023/12/50/refactorings.history
[the remark this answers is not in the document]
Comment (azoitl) [reply org.eclipse.zest.core/.refactorings/2023/12/50/refactorings.history:1]: It looks like all zest projects have this .refactoring folder. I think we should remove this. But in a separate commit.

[the remark this answers is not in the document]
Comment (ptziegler) [reply org.eclipse.zest.core/.refactorings/2023/12/50/refactorings.history:1]: Sounds good to me.

File: org.eclipse.draw2d/src/org/eclipse/draw2d/Label.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (ptziegler) [reply org.eclipse.draw2d/src/org/eclipse/draw2d/Label.java:112]: Forcing a specific logging framework on the user sounds like a bad idea. Then let's just leave it like this.

**Review by ptziegler — APPROVED** [review]

**Pull request merged**